pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -420,10 +463,19 @@ static boolean isExternalViewConverged(Map<String, Map<String, String>> external
       Map<String, String> idealStateInstanceStateMap = entry.getValue();
       for (Map.Entry<String, String> instanceStateEntry : idealStateInstanceStateMap.entrySet()) {
         // Ignore OFFLINE state in IdealState
-        String state = instanceStateEntry.getValue();
-        if (!state.equals(SegmentOnlineOfflineStateModel.OFFLINE) && !state
-            .equals(externalViewInstanceStateMap.get(instanceStateEntry.getKey()))) {
-          return false;
+        String idealStateInstanceState = instanceStateEntry.getValue();
+        if (!idealStateInstanceState.equals(RealtimeSegmentOnlineOfflineStateModel.OFFLINE)) {
+          continue;
+        }
+        String instanceName = instanceStateEntry.getKey();
+        String externalViewInstanceState = externalViewInstanceStateMap.get(instanceName);
+        if (!idealStateInstanceState.equals(externalViewInstanceState)) {
+          if (RealtimeSegmentOnlineOfflineStateModel.ERROR.equals(externalViewInstanceState)) {
+            LOGGER.warn("Found ERROR instance: {} for segment: {}, table: {}", instanceName, segmentName,
```

Comment:
I think this should be error level.

Reply:
I think  a warning is better. The server instance, when it transitioned into ERROR state, would have logged an error anyway. It is not an error in the controller if it discovers that a server was malfunctioning.

On another (but related) note, I would avoid logging strings that have "ERROR". It is quite irritating to see these come up when we see some problem, and want to get all the error logs (as in, logs that have been logged with LOGGER.error() statement). I am not sure we can do much here, but something to note.